platform/src/main/java/org/hillview/sketches/PCACorrelationSketch.java

```diff
@@ -28,29 +28,49 @@
 
 /**
  * This class computes the correlations between different columns in the table.
- * This class is very similar to the SampleCorrelationSketch, apart from two important differences. First, it uses
- * the full data (it doesn't sample). Second, it rescales the columns by using the mean and standard deviation from
- * the BasicColStats's.
+ * This class is very similar to the SampleCorrelationSketch, except that it handles missing values
+ * carefully (it does not include them while computing expectations).
  */
-public class FullCorrelationSketch implements ISketch<ITable, CorrMatrix> {
+
+public class PCACorrelationSketch implements ISketch<ITable, CorrMatrix> {
     private final String[] colNames;
+    private final long seed;
+    private long totalRows;
+    private final double samplingRate;
+
+    public PCACorrelationSketch(String[] colNames, long totalRows, long seed) {
+        this.colNames= colNames;
+        this.samplingRate = Math.min(1, 10000.0*colNames.length/totalRows);
+        this.seed = seed;
+    }
 
-    public FullCorrelationSketch(String[] colNames) {
-        this.colNames = colNames;
+    public PCACorrelationSketch(String[] colNames) {
+        this.colNames= colNames;
+        this.samplingRate = 1;
+        this.seed = 0;
     }
 
     @Override
-    public CorrMatrix create(ITable table) {
+    public CorrMatrix create(ITable data) {
         for (String col : this.colNames) {
-            if ((table.getSchema().getKind(col) != ContentsKind.Double) &&
-                    (table.getSchema().getKind(col) != ContentsKind.Integer))
+            if ((data.getSchema().getKind(col) != ContentsKind.Double) &&
+                    (data.getSchema().getKind(col) != ContentsKind.Integer))
                 throw new InvalidParameterException("Correlation Sketch requires column to be " +
                         "integer or double: " + col);
         }
         CorrMatrix corrMatrix = new CorrMatrix(this.colNames);
+        ITable table;
+        if(this.samplingRate == 1)
```

Comment:
Missing space.
Also, please use >=; in general we try to avoid comparing doubles for equality. But we can't always.